python/test/segy.py

```diff
@@ -2274,3 +2280,133 @@ def test_trace_header_extensions():
 
         assert np.array_equal(f.attributes(TraceField.INLINE_3D), [1, 1])
         assert np.array_equal(f.attributes(TraceField.CROSSLINE_3D), [20, 21])
+
+
+@tmpfiles(testdata / 'mapping-all-types.sgy')
+@tmpfiles(testdata / 'mapping-all-types-lsb.sgy')
+def test_read_write_all_custom_mapping_types(tmpdir):
+    from dataclasses import dataclass
+    import numbers
+
+    @dataclass
+    class HeaderTest:
+        name: str
+        opened: numbers.Real
+        changed: numbers.Real
+
+    types_header0_cases = [
+        HeaderTest("int2", 1, -101),
+        HeaderTest("int4", 2, -102),
+        HeaderTest("int8", 3, -103),
+        HeaderTest("uint2", 4, 104),
+        HeaderTest("uint4", 5, 105),
+        HeaderTest("uint8", 6, 106),
+        HeaderTest("ibmfp", 7.0, -107.0),
+        HeaderTest("ieee32", 8.0, -108.0),
+        HeaderTest("ieee64", 9.0, -109.0),
+        HeaderTest("linetrc", 10, 110),
+        HeaderTest("reeltrc", 11, 111),
+        HeaderTest("linetrc8", 12, 112),
+        HeaderTest("reeltrc8", 13, 113),
+        # scaled values are not scaled. As in segyio 1.x, raw value is returned
+        HeaderTest("coor4", 14, -114),
+        HeaderTest("elev4", 15, -115),
+        HeaderTest("time2", 16, -116),
+        HeaderTest("spnum4", 17, -117),
+        # scale 6 type is inconsistent and not supported well yet
+        HeaderTest("scale6_mant", 18, -118),
+        HeaderTest("scale6_exp", 19, -119),
+        HeaderTest("header_name", b"TYPES\x00\x00\x00", b"TYPES   "),
+    ]
+
+    types_header1_cases = [
+        HeaderTest("int2", -1001, 10001),
+        HeaderTest("int4", -1002, 2000002),
+        HeaderTest("int8", -1003, 3000000003),
+        HeaderTest("uint2", 1004, 4004),
+        HeaderTest("uint4", 1005, 5000005),
+        HeaderTest("uint8", 1006, 60000000006),
+        HeaderTest("ibmfp", -1007.0, 707.125),
+        HeaderTest("ieee32", -1008.0, 808.75),
+        HeaderTest("ieee64", -1009.0, 909.5),
+        HeaderTest("linetrc", 1010, 10010),
+        HeaderTest("reeltrc", 1011, 10011),
+        HeaderTest("linetrc8", 1012, 1000012),
+        HeaderTest("reeltrc8", 1013, 1000013),
+        HeaderTest("coor4", -1014, 100014),
+        HeaderTest("elev4", -1015, 100015),
+        HeaderTest("time2", -1016, 10016),
+        HeaderTest("spnum4", -1017, 100017),
+        # scale 6 type is inconsistent and not supported well yet
+        HeaderTest("scale6_mant", -1018, 1018),
+        HeaderTest("scale6_exp", -1019, 1019),
+        HeaderTest("header_name", b"TYPES\x00\x00\x00", b"TYPES   "),
+    ]
+
+    types_headers = {
+        0: types_header0_cases,
+        1: types_header1_cases,
+    }
```

Comment:
This refers to the (extended?) trace header (=`tracehader_index = 1`) for two traces (`trace[0]` and `trace[1]`). Is this correct?

From the naming, I was initially expecting that we would read the standard trace header (index `0`) and the first extended trace header (index `1`).